docs/serverless-v2/03-03-lambda-processing-phases.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Lambda processing phases
+type: Details
+---
+
+From the moment you create or update a lambda (Function CR) until the time it reaches the final `Running` status phase, the lambda goes through three processing phases:
+
+1. `Initializing`
+2. `Building`
+3. `Deploying`
+
+The diagrams illustrate these three core phases of a lambda processing circle that the Function Controller handles. They also list all custom resources involved in this process and specify in which cases their update is required.
+
+>**NOTE:** Before you start reading, see the [Function CR](#custom-resource-function) document for the custom resource detailed definition, the list of all lambda status phases and reasons for their success or failure.
+
+## Initializing
+
+This initial phase starts when you create a Function CR with configuration specifying the lambda's setup. It ends with creating a ConfigMap and a TaskRun CR ready for building the lambda image.
+
+Updating an existing lambda involves an image rebuild only if any previous lambda processing phase failed (**phase: Failed**), or if you change lambda's configuration, so its body (**function**) or dependencies (**deps**). An update of lambda's environmental variables doesn't require image rebuild, and it only affects KService in the `Deploying` phase.
+
+> **NOTE:** Each time you update lambda's configuration, the Function Controller deletes all previous TaskRun CRs for the given lambda's **UID**.
+
+![Initializing stage](./assets/initializing.svg)
+
+## Building
+
+This phase involves fetching and processing the TaskRun CR. It ends successfully when the lambda image is built and sent to the Docker registry. If the image already existed and only update is required, the Docker image receives a new tag.
+
+![Building stage](./assets/building.svg)
```

Comment:
If the task is still in progress it also means that it is failed?

Reply:
Hm?

Reply:
Ok, I will update it to a stage, not a decision branch.